src/bun.js/webcore/response.zig

```diff
@@ -784,25 +922,78 @@ pub const Fetch = struct {
             return fetch_error.toErrorInstance(this.global_this);
         }
 
-        fn toBodyValue(this: *FetchTasklet) Body.Value {
-            var response_buffer = this.response_buffer.list;
-            this.response_buffer = .{
-                .allocator = default_allocator,
-                .list = .{
-                    .items = &.{},
-                    .capacity = 0,
-                },
+        pub fn onStartStreamingRequestBodyCallback(ctx: *anyopaque) JSC.WebCore.DrainResult {
+            const this = bun.cast(*FetchTasklet, ctx);
+            if (this.http) |http| {
+                http.enableBodyStreaming();
+            }
+            if (this.aborted.load(.Acquire)) {
+                return JSC.WebCore.DrainResult{
+                    .aborted = {},
+                };
+            }
+
+            this.mutex.lock();
+            defer this.mutex.unlock();
+            const size_hint = this.getSizeHint();
+
+            var scheduled_response_buffer = this.scheduled_response_buffer.list;
+            // This means we have received part of the body but not the whole thing
+            if (scheduled_response_buffer.items.len > 0) {
+                this.scheduled_response_buffer = .{
+                    .allocator = default_allocator,
+                    .list = .{
+                        .items = &.{},
+                        .capacity = 0,
+                    },
+                };
+
+                return .{
+                    .owned = .{
+                        .list = scheduled_response_buffer.toManaged(bun.default_allocator),
+                        .size_hint = size_hint,
+                    },
+                };
+            }
+
+            return .{
+                .estimated_size = size_hint,
             };
+        }
+
+        fn getSizeHint(this: *FetchTasklet) u52 {
+            if (this.body_size == .content_length) {
```

Comment:
it would be more idiomatic Zig for this to be a switch statement

Reply:
fixed on https://github.com/oven-sh/bun/pull/4127/commits/ef6989f4118ee842adf88f83c9d2b3ba64f04f0d